### ISSUE TITLE
Hand digs glass faster than wooden and stone pickaxe

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest_game#2585]
Version: 5.1.1

Digging glass by hand is faster than by pickaxe:

* Hand: 0.7s
* Wooden pickaxe: 1.6s
* Stone pickaxe: 1.0s

Moreover, the leaves digging time of wood/stone swords is not better than hand (0.4s each), but this might be deliberate.

**Comment by paramat** [comment]
Ok, but why is this? There may be a reason and this not so easy to resolve.
More effort required from thread author =)

**Comment by An0n3m0us** [comment]
Seems like it wasn't always the case; pickaxes used to be faster.

"It can be mined by hand, but pickaxes are faster." - [Minetest Wiki](https://wiki.minetest.net/index.php?title=Glass&diff=8431&oldid=8393)

**Comment by paramat** [comment]
For glass:
Things that are 'oddly breakable by hand' are broken almost immediately by hand. Using a tool takes longer. This cannot be 'fixed' because 'oddly breakable by hand' must remain almost immediate, but tools acting on certain dig groups must remain having longer times.

For leaves:
'Fixing' this will probably just cause a similar unrealism to occur between 2 other items, it will cause imbalance elsewhere. Because of how dig time mechanics works, much of it will not make sense.
Besides, you are referring to ultra-blunt wood/stone swords, hand being faster is almost reasonable.

So this is not a bug and is very non-trivial.

**Comment by sfan5** [comment]
> This cannot be 'fixed' because 'oddly breakable by hand' must remain almost immediate, but tools acting on certain dig groups must remain having longer times.

What about introducing a new dig group that fits glass as a material better?

**Comment by paramat** [comment]
smashy? =D
Seriously though, i am not convinced that is worth doing for so few nodes, this is a very minor issue after all. There will always be several dig times comparisons that do not make sense.
Besides, realistically, it would be awkward to smash out a pane of glass with a pickaxe, it is possibly faster to do so with your hands.
Overall, this is not an issue and not worth our time.

**Comment by Moth-Tolias** [comment]
that's some real upside-down clown logic there, paramat. soft fleshy human hands shouldn't break a pane of glass faster than swinging a big heavy metal pick into it.

**Comment by paramat** [comment]
You may be right, but it does not alter my actual point in that comment.

**Comment by An0n3m0us** [comment]
Why couldn't we set `oddly_breakable_by_hand` to `1` instead of `3` so that it takes longer to dig by the players hand?

> Paramat
This cannot be 'fixed' because 'oddly breakable by hand' must remain almost immediate

Why must it remain almost immediate? Isn't that what the `dig_immediate` group does?

**Comment by sfan5** [comment]
> Overall, this is not an issue and not worth our time.

Despite your eager closing of this issue, this is <i>your</i> opinion and I haven't seen another coredev weighing in.

In my opinion, fixing this should at least be looked into and if there "[are] several dig times comparisons that do not make sense" they should be looked into, too.
The point of a game goes out the window if balance is too broken.

**Comment by LoneWolfHT** [comment]
> This cannot be 'fixed' because 'oddly breakable by hand' must remain almost immediate

What's the point of `oddly_breakable_by_hand = 1` if you can't use it? @An0n3m0us said glass has it set to 3, which is AFAIK the fastest digtime you could get with that group

**Comment by appgurueu** [comment]
> The point of a game goes out the window if balance is too broken.
Indeed. MTG, and specifically `default`, is seen as a solid base and should be balanced.

I would not set `oddly_breakable_by_hand = 1`, because breaking glass by hand should remain fast. Instead, I agree to the proposal of adding/introducing a group for glass so that the picks break it at least just as fast.

**Comment by An0n3m0us** [comment]
How about `oddly_faster_by_tool` or something? I cannot think of a good name as I'm not sure which nodes it will apply to (other than glass).
Any suggestions on the group name?

**Comment by TumeniNodes** [comment]
A good group name for glass like nodes could be "shatter"
we could have `oddly_breakable_by_hand` which is current but should be set to nodes one would not normally think breakable by hand 

and then add `breakable_by_hand` for nodes which one associates with being breakable by the hand... (such as glass) and things such as leaves, because one can pull leaves off trees, etc.

just a couple thoughts

**Comment by orbea** [comment]
I think its ignoring both swords and axes and use the values for the hand when breaking glass. You can notice how they don't take any wear.

See issue https://github.com/minetest/minetest_game/issues/2650.

I think PR would help here too in making the balance issues more apparent. https://github.com/minetest/minetest_game/pull/2649

**Comment by Wuzzy2** [comment]
I think this *might* actually be an engine issue. I looked at the code:

Glass has:

```
groups = { cracky = 3, oddly_breakable_by_hand = 3 }
```

* Hand digs `oddly_breakable_by_hand=3` in 0.7 seconds and can't dig `cracky=3`.
* Wooden pickaxe can't dig `oddly_breakable_by_hand=3`, and digs `cracky=3` in 1.6 seconds.

Normally, Minetest falls back to the digging times of the hand when the wielded item can't dig. But not here: Since the pickaxe *can* dig `cracky`, it uses the pickaxe dig time for `cracky`, although the hand would be faster with `oddly_breakable_by_hand`.
A possible engine solution would be make sure the fastest hand time is used. But this solution might have other issues as this introduces more non-obvious behavior.


The simple solution would be to set `oddly_breakable_by_hand=2`. This would increase the hand digging time from 0.7 seconds to 2.0 seconds. Which is then worse than the wooden pickaxe. The downside is that players will HATE this.

A better solution would be to make sure the pickaxes can dig glass faster or at least at the same speed than the hand.

Current dig times for cracky=3:

* Wood = 1.6s
* Stone = 1.0s
* (Hand with `oddly_breakable_by_hand=3`): 0.7s
* Bronze = 0.45s
* Steel = 0.4s
* Mese = 0.2s
* Diamond = 0.1s

As you see, the hand is right between stone and bronze atm. Ideally, the dig times of Wood and stone would be buffed to be both below 0.7s. I suggest 0.6s and 0.5s, respectively.

Sadly, I think the only way to achieve this would be with a new group. I say, "sadly" because of compatibility concerns. Mods suddenly face a brand new digging group in MTG after years. This will come unexpected. But with a new group, all dig times can be dealt with properly. But it feels ugly adding a completely new digging group only for glass.

So all solution I can think of introduce their own problems as well. :-(

Maybe this bug might actually be a "won't fix" or "can't fix", at least not in MTG itself. Maybe the engine might be the best solution, after all.
